ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/state/PendingBlocks.java

```diff
@@ -27,24 +27,49 @@
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Collectors;
 
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.RemovalListener;
+
 public class PendingBlocks {
 
-  private final Map<Hash, Block> pendingBlocks = new ConcurrentHashMap<>();
+  // If more than 100 behind, Besu switch to full synchronization mode. 150 because it is possible
+  // to have multiple versions of the same block number
```

Comment:
Can you point me to where this switch happens?

Reply:
the implementation has changed and is no longer based on this behavior. The node does not restart a FullSync if the distance to the head chain is >100 but when it connects to a new peer.